### PR TITLE
Aiden m/module sizes

### LINKED ISSUE #97: [REQUEST] Module Size Data Handling and Module Data Layouts
**Description**
Develop a good way to handle modules with different sizes that's readable from a 2D array. Allow developers to easily design the shape of a module and customize the interiors of each module using prefabs.

### PR DESCRIPTION
**Pull request by rdpolarity**
## Summary
- Added module layout, icon and description serializable properties to ModuleData scriptable object for @TeaRussell 
- Added hull manager to include origins 2D array to determine the source position of a module
- Updated hull module 2D array to include module sizes (generated using the origins array)
- Added hull module walls generator and made it update on the event modules are placed, moved or deleted. 

## Known Problems
- The correct rotation of in game modules don't align correctly to the data in the array and inside the prefabs

## Linked Issues
closes #97

## Screenshots (optional)
https://cdn.discordapp.com/attachments/819755566913814559/840935225563283456/bSDILkbIzU.mp4

### CODE REVIEW

**Comment by MistaMishMash** [comment]
It's probably 100% fine, but I accidentally did this while playing with your branch, and wanted to let you know: 
image

File: Assets/Scripts/Game/HullManager.cs [hunk 1]
```diff
@@ -160,6 +161,50 @@ private void GenerateFloor()
...
+			SetupMasterGroup();
+
+			if (moduleGroup != null)
+				Destroy(moduleGroup.gameObject);
+
+			moduleGroup = new GameObject(nameof(moduleGroup));
+			moduleGroup.transform.position = transform.position;
+			moduleGroup.transform.parent = hullGroup.transform;
+
+			moduleGroup.transform.rotation = Entities.PlayerShipEntity.FindObjectOfType<PlayerShipEntity>().gameObject.transform.rotation;
```
Comment (MistaMishMash) [inline Assets/Scripts/Game/HullManager.cs:180]: Not a problem or anything that you need to fix, but just as an FYI, I think you can go PlayerController.Instance.PlayerShipEntity.

**Review by MistaMishMash — APPROVED** [review]
I think this looks really good! W total transparency; I think I understand the gist of the wall generation logic, but some of it has probably gone over my head. 

Some suggestions (don't need to do but would be nice to have I think )

- You have a HullManager and HullData, but finding what a Hull actually is is a little difficult (I think it mentions somewhere in the body of HullData that it's the layout). Would be great to have that description in the class summary somewhere.
- Would be g if ModuleData, HullData, ect, went into the Data folder.
- Some comments on the more complex parts of wall generation would be nice, just so its easier for someone to look at it, and maintain it if need be. Some other parts of the internal logic are a bit ambiguous (eg what is a MasterGroup?) it's not public so not a huge issue, but, again, would be nice to know what they are for maintainability.
- Also that weird wall issue thing I posted a comment about; probably nothing but just thought I would mention. Also, have we tested this with L shaped pieces, just to check?

Again, just general feedback that I think would make things better, but nothing ground-breaking; only implement if you think it would improve the code & you have time. c: 

Approving it, but good idea to wait for Liam H's review too.

**Comment by rdpolarity** [comment]
> It's probably 100% fine, but I accidentally did this while playing with your branch, and wanted to let you know:
> image

This is part of the "Known problem" section.

File: Assets/Scripts/Game/HullManager.cs (hunk 1 above)
Comment (rdpolarity) [reply Assets/Scripts/Game/HullManager.cs:180]: Thanks, I should've commented that this needed to be re-written / find another way. It was a quick fix for aligning to the direction the ship is currently facing

**Review by Liam-Harrison — APPROVED** [review]
Looks good 👍

**Pull request merged**